src/slam_toolbox_common.cpp

```diff
@@ -247,6 +413,7 @@ void SlamToolbox::publishTransformLoop(
 
   rclcpp::Rate r(1.0 / transform_publish_period);
   while (rclcpp::ok()) {
+    boost::this_thread::interruption_point();
```

Comment:
Ditto on previous, isn't there a cleaner API that `join` triggers we can check if the thread should exit from?